shared/nm-utils/nm-macros-internal.h

```diff
@@ -639,8 +639,11 @@ NM_G_ERROR_MSG (GError *error)
 #define NM_PROPAGATE_CONST(test_expr, ptr) (ptr)
 #endif
 
+/* with the way it is implemented, the caller may or may not pass a trailing
+ * ',' and it will work. However, this makes the macro unsuitable for initializing
+ * an array. */
 #define NM_MAKE_STRV(...) \
```

Comment:
This is not for initializing an array `const char *[]` (it won't even work there) but a pointer to an array `const char *const*`. There are no "initalizers" for that, you'd need to use a C99 compound literal:

```
    .keys = (const char *const[]) { "a", "b", NULL },
```
vs.
```
    .keys = NM_MAKE_STRV ("a", "b"),
```

it requires you to not forget about the NULL sentinel and the second const (which matters here, because it results in the array being in read-only memory).

Reply:
hm. Seems I was wrong, the "const" is not enough to put the pointer into a read-only section (for a static variable). Dunno why not...

Reply:
wrong again. The "const" does make a different :)